ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -169,6 +170,7 @@ public long getLastAppliedIndex() {
   private final AtomicLong inProgressInstallSnapshotRequest;
   private final AtomicLong installedSnapshotIndex;
   private final AtomicBoolean isSnapshotNull;
+  private final AtomicBoolean finishSnapshot;
```

Comment:
Remove finishSnapshot and the related methods since finishSnapshot is not useful.